crates/core/src/generator/templater.rs

```diff
@@ -176,7 +176,7 @@ where
         let value = funcdef
             .value
             .as_ref()
-            .map(|s| self.replace_placeholders(s, placeholder_map))
+            .map(|s| self.replace_placeholders(&s.to_string(), placeholder_map))
```

Comment:
ah, now I remember why `value` is a string in `FunctionCallDefinition` -- it needs to remain a string while it's processed by the templater (here). This line serves to replace placeholders in `value`. For example: `value = "{sendValue}"`

So `FunctionCallDefinition` still needs to be `Option<String>` but `FunctionCallDefinitionStrict` should remain be a U256.